en/sensor_bus/i2c.md

```diff
@@ -1 +1,38 @@
-# I2C Bus Overview
\ No newline at end of file
+# I2C Bus Overview
+
+I2C is a packet-switched serial communication protocol that allows multiple master devices to connect to multiple slave devices using only 2 wires per connection. It is intended for attaching lower-speed peripheral ICs to processors and microcontrollers in short-distance, intra-board communication.
+
+Pixhawk/PX4 support it for:
+* Connecting off board components that require greater data rates than provided by a strict serial UART: e.g. rangefinders.
+* Compatibility with peripheral devices that only support I2C.
+* Allowing multiple devices to attach to a single bus (useful for conserving ports). For example, LEDs, Compass, rangefinders etc.
+
+> **Tip** IMUs (accelerometers/gyroscopes) should not be attached via I2C (typically the [SPI](https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus) bus is used). The bus is not fast enough even with a single device attached to allow vibration filtering (for instance), and the performance degrades further with every additional device on the bus.
+
+
+## Integrating I2C Devices
```

Comment:
@bkueng @mcharleb Looking for advice on this section.
1. Below is supposed to be very high level advice on integrating I2C hardware to work with PX4. But since I have never done it may be incorrect. Is this the right way?
2. Looking at the drivers doc it seems that it might not be. They indicate that the drivers interface is deprecated and instead teh https://github.com/px4/DriverFramework should be used. However I cannot find any I2C examples using the framework.
   - Can you clarify what the rules are
   - Any reason not to pull this readme https://github.com/px4/DriverFramework into the documentation set?
   - What else should we say.
   - What other docs should we point to -https://github.com/px4/DriverFramework ? https://dev.px4.io/en/middleware/drivers.html ?

Reply:
1. Yes it looks good.
2. Initially it was the plan to switch to DriverFramework. But that never really took off, so we recently decided to make sure we can run our existing NuttX drivers on other OSes too. Which means doing the opposite what the current docs say: moving away from DriverFramework (@dagar is working on that).